src/Shared/Microsoft.DotNet.GitHub.Authentication/IGitHubAppTokenProvider.cs

```diff
@@ -0,0 +1,12 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.DotNet.GitHub.Authentication
+{
+    public interface IGitHubAppTokenProvider 
+    {
+        string GetAppToken();
+        string GetAppTokenFromEnvironmentVariableBase64(int gitHubAppId, string environmentVariableName);
```

Comment:
This should probably just be a different implementation of the provider.  I wouldn't personally add this to the interface.  The "DarcBot" should just choose a different IGutHubAppTokenProvider that does the right thing.

Reply:
I have lots of comments about the implementation of that method, but for now, just leaving it off the interface is the best option. :-)

Reply:
I'm confused this was the original comment -> https://github.com/dotnet/arcade-services/pull/1084#discussion_r405812677, I extracted this to an interface. Now I feel the comments are conflicting.

Reply:
Extract the method you want to the new interface. I didn't realize this other weird method exists.  It's not a good abstraction, and you don't need it for testing.  You don't need every method to be in the interface.

Honestly, I'd rather we delete that other method and fix it's implementation to be better encapsulated, but that change doesn't impact the testing PR at all, so it's not really worth it for you to do right now.  Like it doesn't make sense for it to be told an environment variable key for it to pull it out... why didn't the caller just give us the answer instead of binding this method to the environment (especially since the first argument, gitHubAppid, is ALSO pulled from the environment... they pulled one of them but not the other)...  But all of that is sort of silly, because the implementation of this method doesn't use any instance data, it should just be a static method somewhere.   But really really, there should be a "EnvironmentGitHubAppTokenProvider" that implements the new interface, and just implements the "GetAppToken" appropriately.  The whole point of the original class was that the caller didn't need all the gory details... the added method takes all the gory details and destroys the whole point of the class.

TL;DR: Remove the GetAppTokenFromEnvironmentVariableBase64 method, leave the GetAppToken method.

Reply:
The important method in the interface is GetAppToken, right? If I understand what @ChadNedzlek means, I think he's right. The Environment variable method should probably not be in the interface, and there would be a `GithubAppTokenProviderFromEnvironment` implementation.

You can just leave it in the class

Reply:
I am going to create another issue to do this cleanup .